Maintenance: Unify examples

mkuczera opened the pull request:
As discuessed, unified real life examples and examples. Also prepare test setup

rotemmiz commented on package.json:
```diff
@@ -24,20 +24,50 @@
-    "react": ">= 15.4.1",
-    "react-native": ">= 0.40.0"
+    "react": "*",
+    "react-native": "*"
```
The project requires RN>40 IIRC

rotemmiz commented on playground/android/build.gradle:
```diff
@@ -5,7 +5,7 @@ buildscript {
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.3.0'
+        classpath 'com.android.tools.build:gradle:2.2.3'
```
com.android.tools.build:gradle:2.2.3 is VERY old. Let's please move to 3.0.1 or at least stay at 2.3.0

rotemmiz replied:
Detox will not work with that gradle plugin version, it requires 3.x.x

rotemmiz commented on playground/android/gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,5 @@
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-3.3-all.zip
```
Same here please, let's move forward, not backwards.

mkuczera commented:
@rotemmiz i updated the PR. Thanks for reviewing ;)

mkuczera commented:
Strange. Ran it this morning and everything was fine. Deleted the app from the phone and ran it again and it was not able to build. I will take this as a task to to everytime before pushing.
- Added google Repo to buildscripts
- Added package.json run-android
- Set gradle version to 4.1

rotemmiz commented:
One more thing that needs to be verified is what npm packs inside the package with the new structure.
I suggest you run `npm pack` and then `tar -tf react-native-interactable-{version}.tgz`, and adjust `.npmignore` accordingly.

rotemmiz commented:
Thanks!
